### PR TITLE
Warn when using secrets without `--trusted`

### PR DESCRIPTION
**Pull request by helenlyang**
# Summary
This updates the CLI to print a warning if a Truss has secrets defined in its config but is pushed without `--trusted`.

# Testing
## `truss push`
Commands below were for pushing a Truss with a dummy secret in `config.yaml`:
```
secrets:
  dummy_access_token: null
```

Without `--trusted`: `poetry run truss push`
image

With `--trusted`: `poetry run truss push --trusted`
Output:
```
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
✨ Model test-model was successfully pushed ✨

|---------------------------------------------------------------------------------------|
| Your model has been deployed as a draft. Draft models allow you to                    |
| iterate quickly during the deployment process.                                        |
|                                                                                       |
| When you are ready to publish your deployed model as a new version,                   |
| pass `--publish` to the `truss push` command. To monitor changes to your model and    |
| rapidly iterate, run the `truss watch` command.                                       |
|                                                                                       |
|---------------------------------------------------------------------------------------|

🪵  View logs for your deployment at https://app.baseten.co/models/7wl16p0q/versions/qrjrg13/logs
```

### CODE REVIEW

File: truss/cli/cli.py [hunk 1]
```diff
@@ -414,6 +414,14 @@ def push(
     # TODO(Abu): This needs to be refactored to be more generic
     service = remote_provider.push(tr, model_name, publish=publish, trusted=trusted)  # type: ignore
 
+    # Log a warning if using secrets without --trusted.
+    if tr.spec.config.secrets and not trusted:
+        not_trusted_text = """
+Warning: your Truss has secrets but was not pushed with --trusted.
+Please push with --trusted to grant access to secrets.
+"""
+        click.echo(not_trusted_text)
```
Comment (helenlyang) [inline truss/cli/cli.py:423]: Feedback on messaging, formatting, etc. welcome!

I could also stack this PR on top of https://github.com/basetenlabs/truss/pull/717 and use `console.print(not_trusted_text, style=yellow)`.

Comment (squidarth) [reply truss/cli/cli.py:423]: I just merged https://github.com/basetenlabs/truss/pull/717, so you can merge master and use the colors

One other thought here, how does this look if we put it after the "Model was successfully pushed" copy? I worry that it gets lost if it's above the big block of development model text

Comment (helenlyang) [reply truss/cli/cli.py]: Done, I think both changes help!

**Review by squidarth — APPROVED** [review]
lgtm, awesome work @helenlyang! one small note, from the screenshot, the yellow might be a little subtle, it might be better as red. 

another non-blocking this is that it might be good to pull the check into a `config_warnings` function that we could add more warnings too, but I think it's ok to run without that

**Comment by helenlyang** [comment]
Thanks @squidarth! Made the readability changes and updated the screenshot. I added a TODO for the `config_warnings` item--think it might make sense to add that once we have multiple config checks

**Comment by squidarth** [comment]
> TODO for the config_warnings item--think it might make sense to add that once we have multiple config checks

agreed

**Pull request merged**